nix_review/app.py

```diff
@@ -145,25 +146,46 @@ def find_nixpkgs_root() -> Optional[str]:
         prefix.append("..")
 
 
+class DisableKeyboardInterrupt:
+    def __enter__(self) -> None:
+        self.signal_received = False
+        self.old_handler = signal.signal(signal.SIGINT, self.handler)
+
+    def handler(self, sig: Any, frame: Any) -> None:
+        print("Ignore Ctlr-C: Cleanup in progress... Don't be so impatient human!")
+
+    def __exit__(self, type: Any, value: Any, traceback: Any) -> None:
+        signal.signal(signal.SIGINT, self.old_handler)
+
+
 class Worktree:
     def __init__(self, name: str) -> None:
-        self.worktree_dir = os.path.join(f"./.review/{name}")
-        os.makedirs(self.worktree_dir, exist_ok=True)
+        worktree_dir = os.path.realpath(os.path.join("review", name))
+        try:
+            os.makedirs(worktree_dir)
+        except FileExistsError:
+            print(f"{worktree_dir} already exists. Is a different review already running?")
+            raise
+        self.worktree_dir: Optional[str] = worktree_dir
         with tempfile.NamedTemporaryFile() as cfg:
             cfg.write(b"pkgs: { allowUnfree = true; }")
             cfg.flush()
             self.environ = os.environ.copy()
             os.environ["NIXPKGS_CONFIG"] = cfg.name
-            os.environ["NIX_PATH"] = f"nixpkgs={os.path.realpath(self.worktree_dir)}"
+            os.environ["NIX_PATH"] = f"nixpkgs={worktree_dir}"
 
     def __enter__(self) -> str:
+        assert self.worktree_dir is not None
         return self.worktree_dir
 
     def __exit__(self, exc_type: Any, exc_value: Any, traceback: Any) -> None:
-        shutil.rmtree(self.worktree_dir)
-        sh(["git", "worktree", "prune"])
-        os.environ.clear()
-        os.environ.update(self.environ)
+        if self.worktree_dir is None:
+            return
+        with DisableKeyboardInterrupt():
```

Comment:
This can still leave a small window to send a second sigint, but humans are not usually that fast.